pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -260,11 +261,33 @@ func generateStatefulSetWithMinReadySeconds(minReadySeconds int32) *apps.Statefu
 	}
 }
 
+func makeStatefulSetWithMaxUnavailable(maxUnavailable *int) *apps.StatefulSet {
```

Comment:
I would discourage the use of a helper function like this to build test tables. DRY principles don't necessarily apply to tests.
A couple of references, if you are interested in reading more:
https://mtlynch.io/good-developers-bad-tests/
https://marcingryszko.medium.com/good-vs-bad-test-helpers-62d552004bc5